core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -225,6 +225,8 @@ public Expression filter() {
     long splitSize;
     if (options.containsKey(TableProperties.SPLIT_SIZE)) {
       splitSize = Long.parseLong(options.get(TableProperties.SPLIT_SIZE));
+    } else if (options.containsKey(TableProperties.METADATA_SPLIT_SIZE)) {
+      splitSize = Long.parseLong(options.get(TableProperties.METADATA_SPLIT_SIZE));
```

Comment:
I don't think this is necessary. When options are used to set the split size in the Spark reader, it uses `TableProperties.SPLIT_SIZE` to pass it here. That should work for metadata tables as well, right?

The situation that we need to handle in this PR is setting the default, like you had before. We just want to use a metadata split size so you can set it and not have the same split size used (by default) for both metadata and data scans.

Reply:
@rdblue Thanks for the comments and updated the PR accordingly.